### PR TITLE
docs: Release procedure documentation

### PR DESCRIPTION
**Pull request by spydon**
# Description

Just adds documentation about the release procedure and breaking changes.

## Checklist

- [x] The title of my PR starts with a [Conventional Commit] prefix (`fix:`, `feat:`, `docs:` etc).
- [x] I have read the [Contributor Guide] and followed the process outlined for submitting PRs.
- [x] I have updated/added tests for ALL new/updated/fixed functionality.
- [x] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [x] I have updated/added relevant examples in `examples`.

## Breaking Change

- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flame-engine/flame/issues
[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Flame Style Guide]: https://github.com/flame-engine/flame/blob/main/STYLEGUIDE.md
[Conventional Commit]: https://conventionalcommits.org

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -114,6 +123,49 @@ Examples of PR titles:
...
+two versions after the current one, such that there will be at least one stable release where the
+users get to see the deprecation warning and in the version after that (or a later version) the
+deprecated entity should be removed.
+
+
+## Creating a release
+
+There are a few things to think about when doing a release:
+ 
+ - Go through the codebase and look for deprecated methods/fields and remove the ones that are
```
Comment (st-pasha) [inline CONTRIBUTING.md:156]: Perhaps searching for `@Deprecated` is easier

Comment (spydon) [reply CONTRIBUTING.md:156]: Haha I don't think anyone would go through file by file and do it manually, but i can update it if you think is unclear :smile:

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -114,6 +123,49 @@ Examples of PR titles:
...
+
+## Merging a pull request
+
+When merging a pull request, make sure that the title of the merge commit has the correct
+conventional commit tag and a descriptive title. This is extra important since sometimes the title
+of the PR doesn't reflect what GitHub defaults to for the merge commit title (if the title has been
+changed during the life time of the PR for example).
+
+All the default text should be removed from the commit message and if the PR is breaking, the
+instructions from the "Migration instruction" should be copied into the commit message.
```
[the remark this answers is not in the document]
Comment (spydon) [reply CONTRIBUTING.md:139]: It feels like the changelog would be very verbose if we did that, and I think what is currently written in the descriptions isn't always fit for the changelog. I'd prefer to only be verbose on the breaking changes and then if someone is interested in what it said in the description they could just go to the PR.

**Review by erickzanardo — APPROVED** [review]
Excellent! LGTM!

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -114,6 +123,50 @@ Examples of PR titles:
...
  - refactor: Optimize the structure of the game loop
 
+
+# Maintainers
+
+These instructions are for the maintainers of Flame.
+
+## Merging a pull request
+
+When merging a pull request, make sure that the title of the merge commit has the correct
```
Comment (wolfenrain) [inline CONTRIBUTING.md:133]: Should we also have an entry about the fact that a PR should not contain multiple fixes/features? We have overlooked that a few times in the past but it does make it difficult to see it all back in the CHANGELOG

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -114,6 +123,50 @@ Examples of PR titles:
...
+commit message.
+
+
+## Breaking changes
+
+When doing breaking changes a deprecation tag should be added when possible and contain a message
+that conveys to the user what which version that the deprecated method/field will be removed in and 
+what method they should use instead to perform the task. The version specified should be at least
+two versions after the current one, such that there will be at least one stable release where the
+users get to see the deprecation warning and in the version after that (or a later version) the
```
Comment (wolfenrain) [inline CONTRIBUTING.md:149]: The first part of this is general for all contributors imho. And we should maybe give an example

For example:

Lets say the current version is 1.1.2, and the next version that will have the deprecated function is 1.2.0, then the deprecated tag could look like this:

```dart
@Deprecated('Will be removed in 1.3.0')
void deprecatedFeature() {}
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -114,6 +123,50 @@ Examples of PR titles:
...
+
+
+## Creating a release
+
+There are a few things to think about when doing a release:
+ 
+ - Search through the codebase for `@Deprecated` methods/fields and remove the ones that are marked
+   for removal in the version that you are intending to release.
+ - Create a PR containing the changes for removing the deprecated entities.
+ - Run `melos version -V <package>:<version> -V <package>:<version>` for Melos to generate a
```
Comment (wolfenrain) [inline CONTRIBUTING.md:160]: Twice the `-V`? Is it meant to be current vs next? If so we we should make that more explicit in the command

**Review by wolfenrain — CHANGES_REQUESTED** [review]
Left a few comments that I atleast want some clarity on, LGTM apart from that tho

File: CONTRIBUTING.md (hunk 3 above)
Comment (spydon) [reply CONTRIBUTING.md:133]: That's outside of the scope for this section, that should be on the section about making a PR.

File: CONTRIBUTING.md (hunk 5 above)
Comment (spydon) [reply CONTRIBUTING.md:160]: Yes, once for each package, and the version is the version for that package. Would it be more clear if I wrote it like this?
```suggestion
 - Run `melos version -V <package1>:<version> -V <package2>:<version>` for Melos to generate a
```

File: CONTRIBUTING.md (hunk 4 above)
Comment (spydon) [reply CONTRIBUTING.md:149]: That it already defined in the "Breaking changes" section, that section could be improved, but it's also outside the scope of this PR.

Comment (spydon) [reply CONTRIBUTING.md:149]: Oh wait, that's this section. :sweat_smile: 
I thought we had this defined under "Open a pull request" too, this section should be moved to there.

Comment (spydon) [reply CONTRIBUTING.md]: Moved it and added an example.

File: CONTRIBUTING.md (hunk 3 above)
Comment (spydon) [reply CONTRIBUTING.md:151]: Since I already touched that section in this PR now I added a note about that too.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -114,6 +140,41 @@ Examples of PR titles:
...
+
+## Creating a release
+
+There are a few things to think about when doing a release:
+ 
+ - Search through the codebase for `@Deprecated` methods/fields and remove the ones that are marked
+   for removal in the version that you are intending to release.
+ - Create a PR containing the changes for removing the deprecated entities.
+ - Run `melos version -V <package1>:<version> -V <package2>:<version>` for Melos to generate a
+   `CHANGELOG.md` file.
```
Comment (wolfenrain) [inline CONTRIBUTING.md:169]:
```suggestion
   `CHANGELOG.md` file for each package you provide.
```

To make it more clear in general? Cause that was what originally confused it, that it was only for a single package

**Review by wolfenrain — APPROVED** [review]
LGTM, one small nit

File: CONTRIBUTING.md (hunk 6 above)
Comment (spydon) [reply CONTRIBUTING.md:169]: It is also generating a general changelog, maybe "file" should just be changes to "files"?

Comment (spydon) [inline CONTRIBUTING.md:169]:
```suggestion
 - Run `melos version -V <package1>:<version> -V <package2>:<version>` for Melos to generate
   `CHANGELOG.md` files.
```

Comment (wolfenrain) [reply CONTRIBUTING.md:169]: Yea that sounds much better if it does a general one aswell!

**Pull request merged**